.github/workflows/release-cli.yml

```
@@ -0,0 +1,42 @@
+name: Create Github Release with CLI Jar and Distribution
+on:
+  workflow_call:
+    inputs:
+      tag:
+        required: true
+        type: string
+
+jobs:
+  release-job:
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ needs.tag-job.outputs.tag }}
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: 'gradle'
+      - name: Build jar with latest tagged version
+        run: |
+          ./gradlew --build-cache :cli:jar
+          ./gradlew --build-cache :cli:assembleDist
```

Since these tasks only exist in `cli` you don't need to scope them to the `cli` subproject. Also you can run multiple gradle tasks in a single invocation.
```suggestion
          ./gradlew --build-cache jar assembleDist
```